tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -742,3 +271,316 @@ def _enable_disable_volume_profiling(self):
+        try:
+            stored_volume_status = NS._int.client.read(
+                "clusters/%s/Volumes/%s/status" % (
+                    NS.tendrl_context.integration_id,
+                    volumes['volume%s.id' % index]
+                )
+            ).value
+            current_status = volumes['volume%s.status' % index]
+            if stored_volume_status != "" and \
+                current_status != stored_volume_status:
+                msg = "Status of volume: %s " + \
+                      "changed from %s to %s" % (
+                          volumes['volume%s.name' % index],
+                          stored_volume_status,
+                          current_status
+                      )
+                instance = "volume_%s" % volumes[
+                    'volume%s.name' % index
+                ]
+                emit_event(
+                    "volume_status",
+                    current_status,
+                    msg,
+                    instance
+                )
+        except etcd.EtcdKeyNotFound:
+            pass
+
+    volume = NS.gluster.objects.Volume(
+        vol_id=volumes['volume%s.id' % index],
+        vol_type="arbiter"
+        if int(volumes['volume%s.arbiter_count' % index]) > 0
+        else volumes['volume%s.type' % index],
+        name=volumes['volume%s.name' % index],
+        transport_type=volumes['volume%s.transport_type' % index],
+        status=volumes['volume%s.status' % index],
+        brick_count=volumes['volume%s.brickcount' % index],
+        snap_count=volumes['volume%s.snap_count' % index],
+        stripe_count=volumes['volume%s.stripe_count' % index],
+        replica_count=volumes['volume%s.replica_count' % index],
+        subvol_count=volumes['volume%s.subvol_count' % index],
+        arbiter_count=volumes['volume%s.arbiter_count' % index],
+        disperse_count=volumes['volume%s.disperse_count' % index],
+        redundancy_count=volumes['volume%s.redundancy_count' % index],
+        quorum_status=volumes['volume%s.quorum_status' % index],
+        snapd_status=volumes['volume%s.snapd_svc.online_status' % index],
+        snapd_inited=volumes['volume%s.snapd_svc.inited' % index],
+    )
+    volume.save(ttl=SYNC_TTL)
+    rebal_det = NS.gluster.objects.RebalanceDetails(
+        vol_id=volumes['volume%s.id' % index],
+        rebal_id=volumes['volume%s.rebalance.id' % index],
+        rebal_status=volumes['volume%s.rebalance.status' % index],
+        rebal_failures=volumes['volume%s.rebalance.failures' % index],
+        rebal_skipped=volumes['volume%s.rebalance.skipped' % index],
+        rebal_lookedup=volumes['volume%s.rebalance.lookedup' % index],
+        rebal_files=volumes['volume%s.rebalance.files' % index],
+        rebal_data=volumes['volume%s.rebalance.data' % index],
+        time_left=volumes.get('volume%s.rebalance.time_left' % index),
+    )
+    rebal_det.save(ttl=SYNC_TTL)
+    georep_details.save_georep_details(volumes, index)
```

Comment:
Rebal and georep details are node specific here and should be saved by all the nodes. Why by provisioner node only?